pkg/auth/manager.go

```diff
@@ -37,30 +48,22 @@ type authHandler interface {
 type authRequest struct {
 	localIdentity  identity.NumericIdentity
 	remoteIdentity identity.NumericIdentity
-	remoteHostIP   net.IP
+	remoteNodeIP   string
 }
 
 type authResponse struct {
 	expirationTime time.Time
 }
 
 // datapathAuthenticator is responsible to write auth information back to a BPF map
+// Using AuthKey as an argument as the key originates from the datapath,
+// so we do not need to pack/unpack it here.
 type datapathAuthenticator interface {
-	markAuthenticated(result *authResult) error
-}
-
-// authResult contains all relevant information which are necessary to write the info back to data path after a successful authentication.
-type authResult struct {
-	dn             *monitor.DropNotify
-	ci             *monitor.ConnectionInfo
-	localIdentity  identity.NumericIdentity
-	remoteIdentity identity.NumericIdentity
-	remoteNodeID   uint16
-	authType       policy.AuthType
-	expirationTime time.Time
+	markAuthenticated(AuthKey, time.Time) error
+	checkAuthenticated(AuthKey) bool
 }
 
-func newAuthManager(authHandlers []authHandler, dpAuthenticator datapathAuthenticator, ipCache ipCache) (*authManager, error) {
+func newAuthManager(sm signal.SignalRegistrar, authHandlers []authHandler, dpAuthenticator datapathAuthenticator, ipCache ipCache, queueSize int) (*authManager, error) {
```

Comment:
Could you elaborate on this? What are the two aspects that need to be separated here and why? You mean registering the signal Channel to the signal package before calling into this constructor? Would I need to wrap the channel into a struct or can hive track a channel type just like a struct or an interface?

Reply:
yes, i mean the registering the channel to the signal package before calling the constructor. IMO it would be already enough to have this part in the `newManager` function of the cell itself and then callilng the constructor only with the channel.

Reply:
Done, thanks for the suggestion. This allowed the removal of the `SignalRegistrar` interface, as now the unit tests supply a channel directly.